### PR TITLE
Replace created types by existing C types

### PR DESCRIPTION
**Pull request by freekode**
* Replaces defined types to existing one, from C.
* Updated version also added new boards for travis configuration.
* Code reformat
* Separated classes

Locally tested on esp8266

### CODE REVIEW

**Comment by Pillar1989** [comment]
@freekode  Thank you so much. Is there any more code that you don't like?

**Comment by freekode** [comment]
> @freekode Thank you so much. Is there any more code that you don't like?

In future I want to get rid of `Serial` usage in library.
And improve error reporting. Now, only one error code is used in library

**Comment by freekode** [comment]
I will make these changes in next commit. So could you please accept this pull request?

**Comment by Pillar1989** [comment]
I am waiting for your PR, and finally release a new version together

**Comment by freekode** [comment]
Ok, no problem

**Comment by freekode** [comment]
this is my final version

File: examples/basic_demo/basic_demo.ino [hunk 1]
```diff
@@ -31,104 +31,83 @@
 
 #include <Seeed_HM330X.h>
 
-#ifdef  ARDUINO_SAMD_VARIANT_COMPLIANCE
-#define SERIAL_OUTPUT SerialUSB
-#else
-#define SERIAL_OUTPUT Serial
-#endif
-
```
Comment (Pillar1989) [inline examples/basic_demo/basic_demo.ino]: why do you delete this code ? not all arduino printout Serial

**Comment by freekode** [comment]
If you have time, please accept this pull request

**Comment by Pillar1989** [comment]
Your code is only for you, not for anyone else

**Comment by freekode** [comment]
Why this change can't be accepted?

**Comment by Pillar1989** [comment]
Since you are against the original intention of the designer, we hope that this software can be used well on any platform. It shows that you have deleted SerialUSB, which cannot be printed at arduino zero.

**Comment by freekode** [comment]
Oh, this not a problem to revert it, you can just say about it

**Comment by freekode** [comment]
Any other remarks?

File: examples/basic_demo/basic_demo.ino (hunk 1 above)
Comment (freekode) [reply examples/basic_demo/basic_demo.ino]: Returned

**Comment by freekode** [comment]
Returned part which you asked, also added zero board to travis configuration.
Is there any other parts which you don't like?

**Comment by freekode** [comment]
Any other issues with this pull request or it can be accepted without changes?

**Comment by Pillar1989** [comment]
Thanks

**Pull request merged**